### PR TITLE
refact(bdc): use storage in BDC request instead of capacity

### PR DESCRIPTION
**Pull request by prateekpandey14**
**What this PR does / why we need it**:

Commit change the BDC resource request as "storage" instead of
"capacity", to make more generic for a block device.

Respective API changes has been made to https://github.com/openebs/maya/pull/1282  which dynamically provision BDC after the StoragePoolClaim has been created.

Below is the example of `BlockdeviceClaim` schema having a resource request
as storage after bound to the `BlockDevice`.

```yaml
apiVersion: v1
items:
- apiVersion: openebs.io/v1alpha1
  kind: BlockDeviceClaim
  metadata:
    creationTimestamp: "2019-06-12T16:30:42Z"
    finalizers:
    - blockdeviceclaim.finalizer
    generation: 2
    labels:
      openebs.io/storage-pool-claim: sparse-claim-auto
    name: bdc-005779aa-8dda-4531-a5dc-73b79a7daf7c
    namespace: openebs
    ownerReferences:
    - apiVersion: openebs.io/v1alpha1
      blockOwnerDeletion: true
      controller: true
      kind: StoragePoolClaim
      name: sparse-claim-auto
      uid: e7e157e6-90aa-437e-9d6f-345301831193
    resourceVersion: "2327"
    selfLink: /apis/openebs.io/v1alpha1/namespaces/openebs/blockdeviceclaims/bdc-005779aa-8dda-4531-a5dc-73b79a7daf7c
    uid: e3ae1a0b-2f5e-47e1-88bf-8fa990af8cab
  spec:
    blockDeviceName: sparse-37a7de580322f43a13338bf2467343f5
    deviceClaimDetails: {}
    deviceType: ""
    hostName: 127.0.0.1
    resources:
      requests:
        storage: 10G
  status:
    phase: Bound
```

Signed-off-by: prateekpandey14 <prateekpandey14@gmail.com>

### CODE REVIEW

File: pkg/apis/openebs/v1alpha1/blockdeviceclaim_types.go [hunk 1]
```diff
@@ -53,8 +53,8 @@ type DeviceClaimRequirements struct {
 }
 
 const (
-	// ResourceCapacity defines the capacity required as v1.Quantity
-	ResourceCapacity v1.ResourceName = "capacity"
+	// ResourceStorage defines the storage required as v1.Quantity
+	ResourceStorage v1.ResourceName = "storage"
```
Comment (kmova) [inline pkg/apis/openebs/v1alpha1/blockdeviceclaim_types.go]: Can we eliminate the use of this completely and just use the constant defined in K8s itself?

Comment (prateekpandey14) [reply pkg/apis/openebs/v1alpha1/blockdeviceclaim_types.go]: Was thinking the same ...as we are using the same in maya side.

Comment (prateekpandey14) [reply pkg/apis/openebs/v1alpha1/blockdeviceclaim_types.go]: Pushed the changes PTAL

File: integration_tests/k8s/k8sapi.go [hunk 2]
```diff
@@ -104,11 +105,11 @@ func (c k8sClient) ListBlockDeviceClaims() (*apis.BlockDeviceClaimList, error) {
 // NewBDC creates a sample device claim which can be used for
 // claiming a block device.
 func NewBDC(bdcName string) *apis.BlockDeviceClaim {
-	bdcRequirements := apis.DeviceClaimRequirements{
+	bdcRequirements := apis.DeviceClaimResources{
```
Comment (akhilerm) [inline integration_tests/k8s/k8sapi.go]: @prateekpandey14  It would be better to change `bdcRequirements` to `bdcResources`

Comment (prateekpandey14) [reply integration_tests/k8s/k8sapi.go]: done

**Pull request merged**